hoomd/Variant.h

```diff
@@ -50,27 +50,330 @@ class PYBIND11_EXPORT VariantConstant : public Variant
             {
             }
 
+        /// Return the value.
         Scalar operator()(uint64_t timestep)
             {
             return m_value;
             }
 
-        /// Set the value
+        /// Set the value.
         void setValue(Scalar value)
             {
             m_value = value;
             }
 
-        /// Get the value
+        /// Get the value.
         Scalar getValue()
             {
             return m_value;
             }
 
     protected:
-        /// The value
+        /// The value.
         Scalar m_value;
     };
 
+/** Ramp variant.
+
+    Variant that ramps linearly from A to B over a given number of steps.
+*/
+class PYBIND11_EXPORT VariantRamp : public Variant
+    {
+    public:
+
+        /** Construct a VariantRamp.
+
+            @param A The starting value.
+            @param B The ending value.
+            @param t_start The starting time step.
+            @param t_ramp The length of the ramp.
+        */
+        VariantRamp(Scalar A, Scalar B, uint64_t t_start, uint64_t t_ramp)
+            {
+            setA(A);
+            setB(B);
+            setTStart(t_start);
+            setTRamp(t_ramp);
+            }
+
+        /// Evaluate the ramp.
+        Scalar operator()(uint64_t timestep)
+            {
+            if (timestep < m_t_start)
+                {
+                return m_A;
+                }
+            else if (timestep < m_t_start + m_t_ramp)
+                {
+                // interpolate between A and B
+                uint64_t v = timestep - m_t_start;
+                double s = double(v) / double(m_t_ramp);
+                return m_B * s + m_A * (1.0 - s);
+                }
+            else
+                {
+                return m_B;
+                }
+            }
+
+        /// Set the starting value.
+        void setA(Scalar A)
+            {
+            m_A = A;
+            }
+
+        /// Get the starting value.
+        Scalar getA()
+            {
+            return m_A;
+            }
+
+        /// Set the ending value.
+        void setB(Scalar B)
+            {
+            m_B = B;
+            }
+
+        /// Get the ending value.
+        Scalar getB()
+            {
+            return m_B;
+            }
+
+        /// Set the starting time step.
+        void setTStart(uint64_t t_start)
+            {
+            m_t_start = t_start;
+            }
+
+        /// Get the starting time step.
+        uint64_t getTStart()
+            {
+            return m_t_start;
+            }
+
+        /// Set the length of the ramp.
+        void setTRamp(uint64_t t_ramp)
+            {
+            // doubles can only represent integers accuracy up to 2**53.
+            if (t_ramp >= 9007199254740992ull)
```

Comment:
Have to appreciate the caution. :joy:

Reply:
There are lots of subtle considerations that go into enabling 64-bit timesteps.